chips/cc26x0/src/udma.rs

```diff
@@ -0,0 +1,420 @@
+                channel.source_ptr  = base_loc as usize;
+                channel.control.modify(DMATableControl::SRCINC.val(3));
+                channel.control.modify(DMATableControl::DSTINC.val(width as u32));
+                         },    
+        };
+
+        // ------------------------
+        // UDMA Register Section
+        // ------------------------
+
+        let registers: &DMARegisters = unsafe {&*self.regs};
+
+        // Set Bit (Channel N) of CLEARCHNLPRIORITY to set normal priority
+        registers.clear_chnl_priority.set(1<<(dma_channel as u32));
+        // Set Bit (Channel N) of CLEARCHNLPRIALT to use primary channel control
+        registers.clear_chnl_pri_alt.set(1<<(dma_channel as u32));
+        // Set Bit (Channel N) of CLEARBURST to use either single or burst requests
+        registers.clear_burst.set(1<<(dma_channel as u32));
+        // Set Bit (Channel N) of CLEARREQMASK to recognize requests to channel
+        registers.clear_req_mask.set(1<<(dma_channel as u32));
+    }
+
+    
+    pub fn start_xfer(
+        &self,
+        dma_channel: DMAPeripheral,
+    ) {
+        let registers: &DMARegisters = unsafe { &*UDMA.regs };
+        let channel = unsafe{&mut DMACTRLTAB.config_array[dma_channel as usize]};
+        
+        channel.control.modify(DMATableControl::MODE::BASIC);
+        registers.set_channel_en.set(1<<(dma_channel as u32));
+    }
+
+    pub fn prepare_xfer(
+        &self, 
+        dma_channel: DMAPeripheral, 
+        bufptr: usize,
+        xfer_type: DMATransferType,
+        len: usize
+    ) {
+        //let registers: &DMARegisters = unsafe { &*UDMA.regs };
+        let channel = unsafe{&mut DMACTRLTAB.config_array[dma_channel as usize]};
+
+        //make sure `len` isn't longer than the buffer length
+        //find the maximum value `len` could have for the buffer...
+        //let maxlen = buf.len() / match self.width.get() {
+        //        DMAWidth::Width8Bit /*  DMA is acting on bytes     */ => 1,
+        //        DMAWidth::Width16Bit /* DMA is acting on halfwords */ => 2,
+        //        DMAWidth::Width32Bit /* DMA is acting on words     */ => 4,
+        //    };
+
+        //...and set it to this if `len` is larger than the max value
+        //len = cmp::min(len, maxlen);
+
+        //write either the source or destination pointers depending on transfer
+        //type 
+
+        //write the length of the transfer to the channel config
+        channel.control.modify(DMATableControl::TRANSFERSIZE.val(len as u32));
+
+        match xfer_type {
+            DMATransferType::DataCopy => {channel.source_ptr   = bufptr;},
+            DMATransferType::DataTx   => {channel.source_ptr   = bufptr;},
+            DMATransferType::DataRx   => {channel.dest_ptr     = bufptr;},    
+        };
+
+
+        //Enable the transfer complete interrupt
+        //only if this is a software DMA transfer (not necessary if a peripheral transfer)
+
+    }
+
+    pub fn do_xfer(
```

Comment:
`do_transfer`, or `prepare_and_start_transfer` would be even more explicit; I get the impression from `do` that the transfer will be complete once the function returns